src/main/java/de/retest/recheck/auth/RetestAuthentication.java

```diff
@@ -90,45 +98,66 @@ private void login() {
 			final String redirectUri = "http://localhost:" + callback.server.getLocalPort();
 			final String state = UUID.randomUUID().toString();
 
-			final KeycloakUriBuilder builder = deployment.getAuthUrl().clone() //
-					.queryParam( OAuth2Constants.RESPONSE_TYPE, OAuth2Constants.CODE ) //
-					.queryParam( OAuth2Constants.CLIENT_ID, deployment.getResourceName() ) //
-					.queryParam( OAuth2Constants.REDIRECT_URI, redirectUri ) //
-					.queryParam( OAuth2Constants.STATE, state ) //
-					.queryParam( OAuth2Constants.SCOPE, OAuth2Constants.OFFLINE_ACCESS );
+			final URIBuilder builder = new URIBuilder( AUTH_URL );
+			builder.addParameter( "response_type", "code" );
+			builder.addParameter( "client_id", client );
+			builder.addParameter( "redirect_uri", redirectUri );
+			builder.addParameter( "state", state );
+			builder.addParameter( "scope", "offline_access" );
 
 			final URI loginUri = URI.create( builder.build().toString() );
 			handler.showWebLoginUri( loginUri );
 
 			callback.join();
 
 			if ( !state.equals( callback.result.getState() ) ) {
-				final VerificationException reason = new VerificationException( "Invalid state" );
-				handler.loginFailed( reason );
+				handler.loginFailed( new RuntimeException() );
 			}
 
 			if ( callback.result.getError() != null ) {
-				final OAuthErrorException reason =
-						new OAuthErrorException( callback.result.getError(), callback.result.getErrorDescription() );
-				handler.loginFailed( reason );
+				handler.loginFailed( new RuntimeException() );
 			}
 
 			if ( callback.result.getErrorException() != null ) {
 				handler.loginFailed( callback.result.getErrorException() );
 			}
 
-			final AccessTokenResponse tokenResponse =
-					ServerRequest.invokeAccessCodeToToken( deployment, callback.result.getCode(), redirectUri, null );
-			accessToken = tokenResponse.getToken();
+			final TokenBundle bundle = accessCodeToToken( callback.result.getCode(), redirectUri );
+			accessToken = verifier.verify( bundle.accessToken );
```

Comment:
What happens if the access token is null? Looks like this may happen if the response wasn't successful.